api/src/main/java/org/pmiops/workbench/billing/FreeTierBillingService.java

```diff
@@ -295,4 +296,34 @@ public double getUserFreeTierDollarLimit(DbUser user) {
     return Optional.ofNullable(user.getFreeTierCreditsLimitDollarsOverride())
         .orElse(workbenchConfigProvider.get().billing.defaultFreeCreditsDollarLimit);
   }
+
+  /**
+   * Set a custom Free Tier dollar limit override for this user. If this is greater than the user's
+   * total cost, set their workspaces to active. Note: lowering the limit below total cost will not
+   * set the workspaces to inactive. checkFreeTierBillingUsage() will do this as part of the next
+   * cron run.
+   *
+   * @param user the user as represented in our database
+   * @param dollarLimit the US dollar amount, represented as a double
+   * @return whether the user's total cost is now below their limit
```

Comment:
If I see a method starting with `set` that returns a `boolean`, I generally assume it returns true if the value was successfully set. This contract is confusing to me.

Reply:
good point.  We don't need this return value.  It's a void now.

Reply:
Update: it returns a boolean again, and works in the way you describe.  Method is now called maybeSetDollarLimitOverride() and has more internal logic